types-tests/message.test-d.ts

```diff
@@ -4,8 +4,6 @@ import { App, MessageEvent, GenericMessageEvent, BotMessageEvent, MessageReplied
 const app = new App({ token: 'TOKEN', signingSecret: 'Signing Secret' });
 
 expectType<void>(
-  // TODO: Resolve the event type when having subtype in a listener constraint
-  // app.message({pattern: 'foo', subtype: 'message_replied'}, async ({ message }) => {});
```

Comment:
As I mentioned in this comment, `app.message({pattern: 'foo', subtype: 'message_replied'}` or something like that can be another option. I'm not sure if it's better than the type parameter though.